### DISCUSSION TITLE
Annotation Action Support

### DISCUSSION THREAD

**Discussion by bradenhs** [grafana/grafana#42010] [Plugins]
![MicrosoftTeams-image](https://user-images.githubusercontent.com/22484684/143951819-0328820c-6412-4567-b2c1-edbe8bdca724.png)

I'm building a plugin that will automatically populate some annotations as shown above. These annotations correspond to user reported incidents. As you can see there is a link in the screenshot. It'd be great if there was a way to make that link clickable. Or perhaps expose some annotation action button API? Even better would be allowing the rendering of a custom React component in the annotation. Also, is there a way to make annotations read-only? In this circumstance I'm not sure it makes much sense to allow the user to edit the annotation.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
